### ISSUE TITLE
​Error on training with DAdaptSGD.

### ISSUE THREAD

**Issue by cavaniho** [bmaltais/kohya_ss#935]
[Dataset 0]
caching latents.
100%|██████████████████████████████████████████████████████████████████████████████████| 29/29 [00:04<00:00,  6.48it/s]
create LoRA network. base dim (rank): 128, alpha: 64.0
neuron dropout: p=None, rank dropout: p=None, module dropout: p=None
create LoRA for Text Encoder: 72 modules.
create LoRA for U-Net: 192 modules.
enable LoRA for text encoder
enable LoRA for U-Net
preparing optimizer, data loader etc.
use D-Adaptation SGD optimizer | {}
running training / 学習開始
  num train images * repeats / 学習画像の数×繰り返し回数: 870
  num reg images / 正則化画像の数: 0
  num batches per epoch / 1epochのバッチ数: 435
  num epochs / epoch数: 10
  batch size per device / バッチサイズ: 2
  gradient accumulation steps / 勾配を合計するステップ数 = 1
  total optimization steps / 学習ステップ数: 4350
steps:   0%|                                                                                  | 0/4350 [00:00<?, ?it/s]
epoch 1/10
steps:   0%|▏                                                            | 14/4350 [00:10<53:15,  1.36it/s, loss=0.119]11:14:36-405569 INFO     ['tensorboard.exe', '--logdir', 'E:/st']
11:14:36-408574 INFO     Starting tensorboard...
steps:   0%|▎                                                            | 21/4350 [00:14<49:42,  1.45it/s, loss=0.102]Serving TensorBoard on localhost; to expose to the network, use a proxy or pass --bind_all
TensorBoard 2.12.3 at http://localhost:6006/ (Press CTRL+C to quit)
steps:   1%|▎                                                           | 23/4350 [00:15<49:00,  1.47it/s, loss=0.0959]11:14:41-424156 INFO     Opening tensorboard url in browser...
steps:  10%|██████                                                      | 435/4350 [04:12<37:48,  1.73it/s, loss=0.131]
saving checkpoint: E:/lora训练\luke-000001.safetensors

epoch 2/10
steps:  20%|████████████▍                                                 | 870/4350 [08:25<33:40,  1.72it/s, loss=nan]
saving checkpoint: E:/lora训练\luke-000002.safetensors

epoch 3/10
steps:  30%|██████████████████▎                                          | 1305/4350 [11:57<27:54,  1.82it/s, loss=nan]
saving checkpoint: E:/lora训练\luke-000003.safetensors

epoch 4/10
steps:  40%|████████████████████████▍                                    | 1740/4350 [15:29<23:13,  1.87it/s, loss=nan]
saving checkpoint: E:/lora训练\luke-000004.safetensors

epoch 5/10
steps:  50%|██████████████████████████████▌                              | 2175/4350 [19:00<19:00,  1.91it/s, loss=nan]
saving checkpoint: [E:/lora训练\luke-000005.safetensors[](url)](![tmpbtas17xy](https://github.com/bmaltais/kohya_ss/assets/107540026/28b95b30-dc8d-43ff-b34b-0cb450df21eb))

**Comment by bmaltais** [comment]
You should open an issue directly with Kohya sd-script for this issue.
